src/components/tooltip/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { FC, ReactNode, useState, useRef } from "react"
+
+import PopUp from "./popUp"
+
+export enum TooltipPosition {
+  top = "top",
+  topLeft = "topLeft",
+  topRight = "topRight",
+  bottom = "bottom",
+  bottomLeft = "bottomLeft",
+  bottomRight = "bottomRight",
+  left = "left",
+  right = "right",
+}
+
+const spacingPosition = {
+  top: "top",
+  topLeft: "top",
+  topRight: "top",
+  bottom: "bottom",
+  bottomLeft: "bottom",
+  bottomRight: "bottom",
+  left: "left",
+  right: "right",
+}
+
+const oppositePosition = {
+  top: TooltipPosition.bottom,
+  topLeft: TooltipPosition.bottomLeft,
+  topRight: TooltipPosition.bottomRight,
+  bottom: TooltipPosition.top,
+  bottomLeft: TooltipPosition.topLeft,
+  bottomRight: TooltipPosition.topRight,
+  left: TooltipPosition.right,
+  right: TooltipPosition.left,
+}
+
+const getSpaceAroundRect = ({ top, bottom, left, right }) => ({
+  top,
+  left,
+  bottom: window.innerHeight - bottom,
+  right: window.innerWidth - right,
+})
+
+export const calculateTooltipPosition = (
+  container,
+  originalPosition,
+  switchThreshold
+) => {
+  if (!switchThreshold && !container) {
```

Comment:
Can you explain to me the concept with the switching threshold? Like I expect that to be something internal to the tooltip component and not or only with something like a build config configurable to the user. Like the tooltip should try to take another position on itself when there is not enough space...
In the storybook, I saw even in the threshold options. Some very strange options I'd never expect from a tooltip to behave like that. Like it switched sides and broke the page width. Id, in general, expects to take less space and line break the content. (Similar to explained above switching left to right)

Reply:
Yeah, I agree it's a bit clunky. Do you have a suggestion on how to check the size of the content that is not rendered though? Since the tooltip content is not in the DOM I can't just get the size of it and use it as a value...